.github/workflows/SyncAnalyzerTemplateMSBuildVersion.yml

```diff
@@ -0,0 +1,48 @@
+name: Sync Microsoft.Build version in analyzer template with Version.props
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'eng/Versions.props'
+
+jobs:
+  Sync-version:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Update analyzer template version with version from Versions.props
+      shell: pwsh
+      run: |
```

Comment:
We should have some error handling here as well - e.g. when the template config moves for some reason (e.g. renaming) - the error should surface somewhere